modules/programs/bat.nix

```diff
@@ -162,14 +162,15 @@ in {
       # NOTE: run `bat cache --build` in an empty directory to work
       # around failure when ~/cache exists
       # https://github.com/sharkdp/bat/issues/1726
-      home.activation.batCache = hm.dag.entryAfter [ "linkGeneration" ] ''
-        (
-          export XDG_CACHE_HOME=${escapeShellArg config.xdg.cacheHome}
-          verboseEcho "Rebuilding bat theme cache"
-          cd "${pkgs.emptyDirectory}"
-          run ${lib.getExe cfg.package} cache --build
-        )
-      '';
+      home.activation.batCache = mkIf (cfg.syntaxes != { } || cfg.themes != { })
+        (hm.dag.entryAfter [ "linkGeneration" ] ''
+          (
+            export XDG_CACHE_HOME=${escapeShellArg config.xdg.cacheHome}
```

Comment:
Drive-by suggestion: this should ideally use `lib.hm.shell.export`, which I just now noticed doesn't do any form of escaping, unlike `lib.strings.toShellVar`...